utils/src/main/kotlin/OrtProxySelector.kt

```diff
@@ -153,6 +159,8 @@ class OrtProxySelector(private val fallback: ProxySelector? = null) : ProxySelec
     override fun select(uri: URI?): List<Proxy> {
         requireNotNull(uri)
 
+        if (uri in noProxyUrls) return NO_PROXY_LIST
```

Comment:
Did you consider that the `uri` may contain a `protocol` or a `port` or `user`, `password` and such?

Reply:
Good point, turns out that constructing / comparing a URI is quote nasty:

E.g. `java.net.URI("foo.bar.com")` evaluates to

```
result = {URI@2908} "foo.bar.com"
 scheme = null
 fragment = null
 authority = null
 userInfo = null
 host = null
 port = -1
 path = "foo.bar.com"
 query = null
 schemeSpecificPart = "foo.bar.com"
 hash = 0
 decodedUserInfo = null
 decodedAuthority = null
 decodedPath = null
 decodedQuery = null
 decodedFragment = null
 decodedSchemeSpecificPart = null
 string = "foo.bar.com"
```

but `java.net.URI("foo.bar.com:8080")` evaluates to

```
result = {URI@2964} "foo.bar.com:8080"
 scheme = "foo.bar.com"
 fragment = null
 authority = null
 userInfo = null
 host = null
 port = -1
 path = null
 query = null
 schemeSpecificPart = "8080"
 hash = 0
 decodedUserInfo = null
 decodedAuthority = null
 decodedPath = null
 decodedQuery = null
 decodedFragment = null
 decodedSchemeSpecificPart = null
 string = "foo.bar.com:8080"
```

!

Reply:
Hm, that's indeed tricky.
Are the results only odd if there is no `scheme`?

Reply:
Looks like so, but I don't want to enforce a scheme, which is why my new version of this PR is based on (sub-)string comparison instead.